lib/lighthouse/benefits_documents/service.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require 'common/client/base'
+require 'lighthouse/benefits_documents/configuration'
+require 'lighthouse/benefits_documents/service_exception'
+require 'lighthouse/service_exception'
+
+module BenefitsDocuments
+  class Service < Common::Client::Base
+    configuration BenefitsDocuments::Configuration
+    STATSD_KEY_PREFIX = 'api.benefits_documents'
+    STATSD_UPLOAD_LATENCY = 'lighthouse.api.benefits.documents.latency'
+
+    def initialize(icn)
+      @icn = icn
+      raise ArgumentError, 'no ICN passed in for LH API request.' if icn.blank?
+
+      super()
+    end
+
+    def queue_document_upload(params, lighthouse_client_id = nil)
+      start_timer = Time.zone.now
+      claim_id = params[:claimId]
+      raise Common::Exceptions::RecordNotFound, claim_id unless claim_id
+
+      jid = submit_document(params[:file], claim_id, params[:trackedItemId], params[:documentType], params[:password],
+                            lighthouse_client_id)
+      StatsD.measure(STATSD_UPLOAD_LATENCY, Time.zone.now - start_timer, tags: ['is_multifile:false'])
+      cleanup_after_upload
+      jid
+    end
+
+    def queue_multi_image_upload_document(params, lighthouse_client_id = nil)
+      start_timer = Time.zone.now
+      claim_id = params[:claimId]
+      raise Common::Exceptions::RecordNotFound, id unless claim
+
+      file_to_upload = generate_multi_image_pdf(params[:files])
+      jid = submit_document(file_to_upload, claim_id, params[:trackedItemId], params[:documentType],
+                            params[:password], lighthouse_client_id)
+      StatsD.measure(STATSD_UPLOAD_LATENCY, Time.zone.now - start_timer, tags: ['is_multifile:true'])
+      cleanup_after_upload
+      jid
+    end
+
+    # Called from worker after queued. Do not call this function directly, use queue methods
```

Comment:
Since the setup is currently async, I wasn't sure what the best way would be to indicate consumers of this service should only upload using the `queue_*` functions, and not call `upload_document` directly. `upload_document` still needs to be public because the worker calls it. Hopefully the comment deters future consumers

Reply:
One way we might consider to make this distinction more explicit is to make a tiny `BenefitsDocuments::WorkerService` class with basically only this method in it. Looks like it makes a post directly with no logic involved. The configuration is a singleton, so if you include it in the `WorkerService` class the way you did here, all the settings should be preserved.

Just a thought. What do you think?

Reply:
That seems like a reasonable way of handling it. I haven't seen it done that way in other places in the codebase, but it's the only way I can think of to ensure the only public methods are the ones we want.

Reply:
Updated. This is a clearer way of distinguishing the difference, great suggestion

Reply:
Sorry that I'm late to the party, but one thing we might consider is if we even want to use jobs for this. The reason I ask is because I think that I remember seeing that Lighthouse was doing their own requeing within their service. It is probably worth discussing with Lighthouse if it is worth doing retries on our side

Reply:
From slack, Michael Harlow said

> [VA.gov](http://va.gov/) should continue to use the retry process. The EVSS and Lighthouse APIs will retry delivery attempts to eFolder. However, [va.gov](http://va.gov/) should use exponential backoff for delivery to Lighthouse.

That reads to me like we should continue using this in a job